packages/frontend/core/src/components/pure/workspace-slider-bar/favorite/add-favourite-button.tsx

```diff
@@ -8,16 +8,48 @@ import { usePageHelper } from '../../../blocksuite/block-suite-page-list/utils';
 
 type AddFavouriteButtonProps = {
   workspace: Workspace;
+  pageId?: string;
 };
 
-export const AddFavouriteButton = ({ workspace }: AddFavouriteButtonProps) => {
+export const AddFavouriteButton = ({
+  workspace,
+  pageId,
+}: AddFavouriteButtonProps) => {
   const { createPage } = usePageHelper(workspace);
   const { setPageMeta } = usePageMetaHelper(workspace);
-  const handleAddFavorite = useAsyncCallback(async () => {
-    const page = createPage();
-    await page.waitForLoaded();
-    setPageMeta(page.id, { favorite: true });
-  }, [createPage, setPageMeta]);
+  const handleAddFavorite = useAsyncCallback(
+    async e => {
+      if (pageId) {
+        e.stopPropagation();
+        e.preventDefault();
+        const page = createPage();
+        await page.load();
+        const parentPage = workspace.getPage(pageId);
+        if (parentPage) {
+          await parentPage.load();
+          const text = parentPage.Text.fromDelta([
+            {
+              insert: ' ',
+              attributes: {
+                reference: {
+                  type: 'Subpage',
```

Comment:
This might be a good idea!

I just started looking at this project and didn't understand the working principle of the entire editor, so I searched for related implementations and reused them.

I'm not sure the best way to implement it, hope someone can help optimize it.

Reply:
@pengx17 I think the concept of `LinkedPage` and `Subpage` are both in use and aligns with our product design, but their APIs can be better encapsulated by the framework. Track in https://github.com/toeverything/blocksuite/issues/5543

Reply:
Added a new commit: [change reference type to LinkedPage](https:github.com/toeverything/AFFiNE/pull/5128/commits/c493b747798bc384c32044596c95470859b5a131)

But a better way should wait https://github.com/toeverything/blocksuite/issues/5543